cmd/tasks_run.go

```diff
@@ -0,0 +1,180 @@
+
+	//TODO: a lot of this will likely be replacable by library functions
+	lagoonConditionalEvaluationEnvironment := tasklib.TaskEnvironment{}
+	//pull all pod env vars
+	if pullWiderEnvironment {
+		allEnvVarNames := os.Environ()
+		for _, n := range allEnvVarNames {
+			kv := strings.Split(n, "=")
+			lagoonConditionalEvaluationEnvironment[kv[0]] = kv[1]
+		}
+	}
+
+	projectVars := []lagoon.EnvironmentVariable{}
+	envVars := []lagoon.EnvironmentVariable{}
+	// get the project and environment variables
+	projectVariables = helpers.GetEnv("LAGOON_PROJECT_VARIABLES", projectVariables, true)
+	environmentVariables = helpers.GetEnv("LAGOON_ENVIRONMENT_VARIABLES", environmentVariables, true)
+	json.Unmarshal([]byte(projectVariables), &projectVars)
+	json.Unmarshal([]byte(environmentVariables), &envVars)
+	lagoonEnvVars := lagoon.MergeVariables(projectVars, envVars)
+
+	// Give context in the logs to how the tasks execution is being evaluated
+	if len(lagoonEnvVars) > 0 {
+		for _, envVar := range lagoonEnvVars {
+			lagoonConditionalEvaluationEnvironment[envVar.Name] = envVar.Value
+		}
+	}
+	return lagoonConditionalEvaluationEnvironment, nil
+}
+
+func evaluateWhenConditionsForTaskInEnvironment(environment tasklib.TaskEnvironment, task lagoon.Task) (bool, error) {
+
+	if len(task.When) == 0 { //no condition, so we run ...
+		return true, nil
+	}
+	fmt.Println("Evaluating task condition - ", task.When)
+	ret, err := tasklib.EvaluateExpressionsInTaskEnvironment(task.When, environment)
+	if err != nil {
+		fmt.Println("Error evaluating condition: ", err.Error())
+		return false, err
+	}
+	retBool, okay := ret.(bool)
+	if !okay {
+		err := fmt.Errorf("Expression doesn't evaluate to a boolean")
+		fmt.Println(err.Error())
+		return false, err
+	}
+	return retBool, nil
+}
+
+func runCleanTaskInEnvironment(incoming lagoon.Task) error {
+	task := lagoon.NewTask()
+	task.Command = incoming.Command
+	task.Namespace = namespace
+	task.Service = incoming.Service
+	task.Shell = incoming.Shell
+	task.Container = incoming.Container
+	err := lagoon.ExecuteTaskInEnvironment(task)
+	return err
+}
+
+func init() {
+	configCmd.AddCommand(tasksRun)
+	tasksRun.Flags().StringVarP(&projectVariables, "project-variables", "v", "",
+		"The projects environment variables JSON payload")
+	tasksRun.Flags().StringVarP(&environmentVariables, "environment-variables", "V", "",
+		"The environments environment variables JSON payload")
+	tasksRun.Flags().StringVarP(&lagoonYml, "lagoon-yml", "l", ".lagoon.yml",
+		"The .lagoon.yml file to read")
+	tasksRun.Flags().BoolVarP(&runPreRollout, "pre-rollout", "", false,
+		"Will run pre-rollout tasks if true")
+	tasksRun.Flags().BoolVarP(&runPostRollout, "post-rollout", "", false,
+		"Will run post-rollout tasks if true")
+	tasksRun.Flags().StringVarP(&namespace, "namespace", "n", "",
+		"The environments environment variables JSON payload")
+	tasksRun.Flags().BoolVarP(&outOfClusterConfig, "out-of-cluster", "", false,
```

Comment:
Could this be inverted logic so no flag is required?

We can update the way builds start in the future to create a `KUBECONFIG` instead of doing it the way we do currently (see: https://github.com/uselagoon/lagoon/blob/main/images/kubectl-build-deploy-dind/build-deploy.sh#L54-L59)

Reply:
Have it check `KUBECONFIG`, if not found, fall back to checking the deployer token, if that isn't found then exit with error?